### PR TITLE
Migrate Reactions store to Sync by default

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Migrate Reactions store to Sync by default

This is the first example of a layered store to be migrated to the new Sync by
default approach

### CODE REVIEW

**Comment by crspeller** [comment]
@jespino Re-add reviews when ready for review.

**Comment by jespino** [comment]
@crspeller Is ready for review, but I don't want people to accidentally merge it before the enterprise PR is merged, that the reason i added the Awaiting PR

File: app/export.go [hunk 1]
```diff
@@ -407,17 +407,15 @@ func (a *App) buildPostReplies(postId string) (*[]ReplyImportData, *model.AppErr
...
+		return nil, err
 	}
 
-	reactions := result.Data.([]*model.Reaction)
-
 	for _, reaction := range reactions {
-		user, err := a.Srv.Store.User().Get(reaction.UserId)
-		if err != nil {
-			return nil, err
+		user, err2 := a.Srv.Store.User().Get(reaction.UserId)
```
Comment (lieut-data) [inline app/export.go]: Why `err2` here vs. predeclaring `user` as done elsewhere?

Comment (jespino) [reply app/export.go]: Yes, I don't have a good reason for that, I think is better all the way around. I going to change it

Comment (jespino) [reply app/export.go]: Fixed

**Pull request merged**